### PR TITLE
Added type caster

### PR DESCRIPTION
**Pull request by dovgopoly**

### CODE REVIEW

File: contracts/libs/utils/BytesCaster.sol [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+library BytesCaster {
```
Comment (Arvolear) [inline contracts/libs/utils/BytesCaster.sol:4]: Please rename this to TypeCaster and add several other cool casting functions

File: contracts/libs/arrays/ArrayHelper.sol [hunk 2]
```diff
@@ -103,6 +103,14 @@ library ArrayHelper {
         array_[0] = elem_;
     }
```
Comment (Arvolear) [inline contracts/libs/arrays/ArrayHelper.sol:104]: Should these 3 functions be in a TypeCaster?

Comment (dovgopoly) [reply contracts/libs/arrays/ArrayHelper.sol]: done

File: contracts/libs/utils/TypeCaster.sol [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ *  @notice This library simplifies non-obvious type castings
+ */
+library TypeCaster {
+    /**
+     *  @notice The function that casts the list of `X`-type elements to the list of uint256
+     *  @param from the list of `X`-type elements
+     *  @return array_ the list of uint256
+     */
+    function asUint256Array(
+        bytes32[] memory from
```
Comment (Arvolear) [inline contracts/libs/utils/TypeCaster.sol:14]:
```suggestion
        bytes32[] memory from_
```

Comment (dovgopoly) [reply contracts/libs/utils/TypeCaster.sol]: done

**Review by Arvolear — APPROVED** [review]
Ok

**Pull request merged**